src/TiciTacaToeyGameEngine.ts

```diff
@@ -277,16 +390,56 @@ class TiciTacaToeyGameEngine implements GameEngine {
         };
         break;
       }
+      case MessageTypes.PLAYER_TIMEOUT: {
+        const game = this.games[message.gameId];
+        const nextPlayer = calculateNextTurn(game);
+        this.games = {
+          ...this.games,
+          [game.gameId]: {
+            ...game,
+            turn: nextPlayer,
+          },
+        };
+
+        let count = 0;
+        let winner = "";
+        game.players.forEach((player) => {
+          if (game.timers[player].timeLeft > 0) {
+            count++;
+            winner = player;
+          }
+        });
+
+        if (count == 1) {
+          this.games = {
+            ...this.games,
+            [message.gameId]: {
+              ...this.games[message.gameId],
+              status: GameStatus.GAME_WON_BY_TIMEOUT,
+              winner: winner,
+              turn: "",
+            },
+          };
+        }
+        break;
+      }
       case MessageTypes.MAKE_MOVE: {
         const game = this.games[message.gameId];
+
+        game.timers[message.playerId].stop();
```

Comment:
Refer to suggestion in Timer class and send increment as an input to stop method